### PR TITLE
Add ipv6 support to port: esp-idf: libcoap

### PR DESCRIPTION
**Pull request by MarkoPura**

### CODE REVIEW

File: src/Kconfig.coap [hunk 1]
```diff
@@ -143,3 +143,8 @@ config GOLIOTH_COAP_MAX_PATH_LEN
     help
         Maximum length of a CoAP path (everything after
         "coaps://coap.golioth.io/").
+
+config LIBCOAP_ENABLE_IPV6_SUPPORT
```
Comment (sam-golioth) [inline src/Kconfig.coap:147]: Since this is specific to ESP-IDF, let's keep in the port folder (`port/esp_idf/components/golioth_sdk/Kconfig`)

**Comment by sam-golioth** [comment]
Okay, I know I said put in the `golioth_sdk` folder, but now I think it should go in a new Kconfig file in the `port/esp_idf/components/coap` folder.

**Comment by MarkoPura** [comment]
While I was pasting it, I thought the same thing.

**Review by sam-golioth — APPROVED** [review]
Thanks @MarkoPura!

**Pull request merged**